common/src/com/thoughtworks/go/domain/materials/git/GitCommand.java

```diff
@@ -299,7 +299,7 @@ private String remoteBranch() {
 
     public void fetch(ProcessOutputStreamConsumer outputStreamConsumer) {
         outputStreamConsumer.stdOutput("[GIT] Fetching changes");
-        CommandLine gitFetch = git(environment).withArgs("fetch", "origin").withWorkingDir(workingDir);
+        CommandLine gitFetch = git(environment).withArgs("fetch", "origin", "--prune").withWorkingDir(workingDir);
```

Comment:
@varshavaradarajan — we only track one remote branch, why is this required then?